### ISSUE TITLE
Use Standard Date/Time stamp and Commit ID for GUI

### ISSUE THREAD

**Issue by ghost** [desktop/desktop#14611]
### The feature request

I love the History, but what is very useless in the gui is how the History of anything is presented.  Currently they are using the "Friendly" date format, which is vague and ambiguous, ie: "yesterday, few minutes ago, 2hours ago".
This is for both GitHub Desktop and also for the web application.
Additionally, in the Desktop, it would be great if we could also see the Commit ID in the list view.  It would be an additional bonus if we could maybe see the parent commits links

In order to see the time I made the commit/change, I have to try to find the "sweet spot" for where to hover over (the whole list item is not triggering the hover...)
![image](https://user-images.githubusercontent.com/62229442/168337284-a8fbee8b-f60e-44b7-b33f-8d84c0299d22.png)

When something goes wrong, we (developers) usually have a pretty good idea on the time we made a particular change, and if we do commit frequently, in order to find the right history to revert to: we literally have to go through a lot of list items that will have the identical friendly date.
If I could see the actual time stamp, it would drastically cut down on how many items I need to review to find the right item.

### Proposed solution

Instead of using the friendly date format, either make it customizable where the user could set a preference for date format, or just change the friendly date format to Standard date format.
I would love it if I could choose to see any date related data in the GUI in this format: (in **BLACK** font and  NOT some light color font that barely separates from the background...)
**05/10/2022 22:44:17**
Can you please consider making this change or add a settings where we could choose date format?
I think tons of us, end-users would really appreciate some ability to customize the interface to what works for us.
![image](https://user-images.githubusercontent.com/62229442/168339547-d51e010b-34ed-4ee1-b65e-a599e54cec47.png)

**Comment by a2937** [comment]
I don't personally wanna add the commit id of the parent to the entry. But I do at least wanna add the commit id right next to the date time. And unfortunately the friendly datetime is super helpful for me.

**Comment by rtapella** [comment]
I'd also like the actual time because the "friendly" datetime is not useful when cross-comparing to other systems.

**Comment by rtapella** [comment]
I didn't even realize you could mouseover the relative date to show a full stamp. I'd still prefer to see it all in the list to cross-compare to lists of time stamps.

**Comment by steveward** [comment]
Requested in https://github.com/desktop/desktop/issues/17702.

**Comment by mattlogic** [comment]
I too would much rather see actual date/time for commits and actions. The current "friendly" format is too vague for comparing with other git clients, logs or file timestamps. "6 months ago" is OK if you just want a rough idea of when something was done, but is practically useless if you need to know whether a particular commit was done before/after something else.

The format for the actual date/time should preferably be selectable from a few common styles, or possibly based on the user's locale. The mouse-over date tooltips are currently US-style like "Dec 8, 2023 5:16 PM GMT" but at least that's unambiguous and readable. Other formats would be YYYY-MM-DD or DD-MMM-YYYY and 24-hour time.

**Comment by gjsmith66** [comment]
Not showing actual date/time makes no sense.  I have no other tools/apps that I use that default to this view.  At least give an option.  Thanks!

**Comment by aterbo** [comment]
I would love to see this option, too, both on web and desktop. I don't know why this was closed. I'm trying to determine what happened when in my code base, and looking at dozens of commits that all say "committed 2 months ago" is useless.

Is there an open issue for this? This issue was closed with no apparent resolution, and the other associated issue #17702 was closed because they were referred to this issue.

**Comment by cfrupp** [comment]
> I don't personally wanna add the commit id of the parent to the entry. But I do at least wanna add the commit id right next to the date time. And unfortunately the friendly datetime is super helpful for me.

In Desktop, it should be configurable by the user: "friendly" or exact, whatever floats your boat.

**Comment by cfrupp** [comment]
The mouse-over is OK, but it's really clumsy if there are a whole lot of commits displayed as "3 days ago". Then you have to mouse around and hover over each one in turn. It would be more efficient if you could see them all at a glance.

**Comment by SRNissen** [comment]
I would love to add my voice to the chorus clamoring for accurate timestamps.

I would add that "no timestamp at all" is a better option than "friendly timestamp" given the frequency of erroneous friendly timestamps as soon as you're more than an hour back, though possibly that should be a separate issue.

**Comment by SRNissen** [comment]
C.f. issue https://github.com/desktop/desktop/issues/18009 which records a UX bug caused by """friendly""" time stamps.

**Comment by steveward** [comment]
From https://github.com/desktop/desktop/issues/18136:

>I find that i nearly always hover over the "X days ago" to get the timestamp for a commit. This is pretty key IMO for a fast moving team that makes multiple commits per day. For example, a deploy may be triggered by a commit, and i'd like to be able to quickly match up exactly which of today's deploys were caused by which commit, or quickly scan for which commits happened before/after a particular deploy.

**Comment by hdhalter** [comment]
I would also prefer to see the actual date. And now I can't even see the actual date when I hover over the relative date and it's driving me crazy! I have to manually count back on my wall calendar or guess the actual date. Any tips on how I can get this ability back?

**Comment by steveward** [comment]
@hdhalter I'm not seeing any issues with hovering over dates on macOS and GitHub Desktop v3.3.9. What version of GitHub Desktop and operating system are you using?

**Comment by hdhalter** [comment]
Hi @steveward - I use a browser, not Desktop, and my mac version is Sonoma 14.3 (23D56). I'm being asked to update to Sonoma 14.3.1 so as soon as I reboot, I'll get the update. Perhaps that will fix it? In the meantime, I see specific dates on some of the comments this issue. So much more helpful!

**Comment by steveward** [comment]
@hdhalter this issue tracker is specifically for GitHub Desktop, if you're having issues with github.com you'll want to reach out to [GitHub Support](https://support.github.com/).

**Comment by steveward** [comment]
Requested in https://github.com/desktop/desktop/issues/18273.

**Comment by thewhistler1** [comment]
Yes, please show actual date.

**Comment by subfuzion** [comment]
Two years is just too long to wait.

**GitHub Time v2** to the rescue (scroll down the page to find the v2 bookmarklet).  
https://justinnoel.dev/2020/11/21/displaying-real-times-in-github/

You have to click the bookmarklet on the toolbar to activate for every page, but at least it's something. Why wait for the heat death of the universe? ;)

<img width="675" alt="image" src="https://github.com/desktop/desktop/assets/159738/03acda42-39c7-4cec-b575-c804d7de5fdb">

**Comment by SRNissen** [comment]
This issue is for the desktop client

**Comment by subfuzion** [comment]
@SRNissen Oops, my bad -- too many tabs open following related threads! Thank you!

**Comment by trusovn** [comment]
I'd also would much rather see the dates. I need to perform cherry-picking from lots of other commits and being able to see the dates would simplify the task.

**Comment by tidy-dev** [comment]
Requested in https://github.com/desktop/desktop/issues/18442

**Comment by Qf4ctor** [comment]
The only thing I don't agree with from [kahopkin](https://github.com/kahopkin)'s proposal is the removal of the author of the commit. Knowing who has made the commit is also useful.

**Comment by tidy-dev** [comment]
Requested in https://github.com/desktop/desktop/issues/18443

**Comment by paul-fastly** [comment]
Yes, please provide at least an option to show the actual date and time instead of the "friendly" reference, in both the desktop and web GUIs.

**Comment by Danie-Brink-PPIT** [comment]
Plase add this option: trying to compare commits on different systems is really painful with the relative date format.

**Comment by pcjbird** [comment]
Why this feature not supported today?

**Comment by antonydandrea** [comment]
This is ridic that this is an issue in 2024, or should I say "0 days ago"

**Comment by SRNissen** [comment]
@steveward I notice this has lived for a while and your team has never rejected it, so I assume it's a prioritization issue.

Are you open to outside PRs on this?

**Comment by Devanshjain2110** [comment]
git show --no-patch --format=%ci {commit}

This command should help for someone who needs this urgently.

P.S : - I would also appreciate the above feature

**Comment by steveward** [comment]
Requested in https://github.com/desktop/desktop/issues/18959.

**Comment by johnmgithub** [comment]
Seems odd that people asking for more than [four years](https://github.com/desktop/desktop/issues/9147) for something that is user-hostile to be changed is met with complete indifference.

**Comment by aterbo** [comment]
> Seems odd that people asking for more than [four years](https://github.com/desktop/desktop/issues/9147) for something that is user-hostile to be changed is met with complete indifference.

@johnmgithub  At this point I don't expect this to actually be resolved, but I'm still subscribed to notifications on this thread for morbid curiosity as to why it is somehow seen as a controversial request. 

I haven't seen a single outside user of these products show opposition to making this a changeable setting. I definitely haven't seen someone clearly state a preference for the current UI display!

**Comment by cfrupp** [comment]
I haven't seen any justification for NOT making this change. On the other hand, I don't have the patience to read through a 4-year history. Does ANYONE have ANY idea why Microsoft has not implemented this request?

**Comment by jartf** [comment]
> Does ANYONE have ANY idea why Microsoft has not implemented this request?

@cfrupp From #9147:
> this has been a longstanding GitHub convention

read: "it works so don't fix it", and

> We haven't heard from many people and that the specific date is a super important piece of information

Basically they're saying there's no popular demand. This issue is proving otherwise though, so someone might wanna look into it.

**Comment by cfrupp** [comment]
So let me get this straight. GitHub is the world's premier change management system, and the GitHub team is afraid of change? Just sayin'.

**Comment by chriswm** [comment]
> Basically they're saying there's no popular demand. This issue is proving otherwise though, so someone might wanna look into it.

How do I add my vote of support for getting this obtuse date format fixed?

I need to check the specific date and time the PR was merged. That should be a 2 second effort, and this lack of precise date is turning a 2 second effort into a 20 minute effort.

**Comment by Jamie-Landeg-Jones** [comment]
> I need to check the specific date and time the PR was merged. That should be a 2 second effort, and this lack of precise date is turning a 2 second effort into a 20 minute effort.

I just found out (from earlier in this thread) that when you hover over the date, the absolute date is shown.

It's a pain to use properly, but it may help you in some cases.

**Comment by Jamie-Landeg-Jones** [comment]
I remember when forums started using this "bling" date format. I think it was a JS script made available to look cool.

I hated it then, but never thought a site like github would pop up with the same imprecise format. Yet, here we are, and here we STILL are.

It's added code to make things worse.

**Comment by winzig** [comment]
(I've been creating forum software for 30 years.) The relative date format is not unique to JS, and the benefit of it is that it's easier for humans to read no matter which timezone they're reading from, or which timezone it was posted in. 

However, while it works well in forums and social media posts, IMO it's a bad choice for precise work like a Git history. At the very least, if it were me, I'd give people a choice (absolute/relative), and I would have it switch from relative to absolute date format when the amount of time being represented is more than a day. "5 min ago" and "2 hrs ago" seems fine, but when I scroll down and see "2 months ago" on different commits that happened days apart, it falls apart...

**Comment by chriswm** [comment]
>I just found out (from earlier in this thread) that when you hover over the date, the absolute date is shown.
>
> It's a pain to use properly, but it may help you in some cases.

I eventually stumbled upon this too. It's not available in all places where the time is shown though, and the first couple places I tried to hover / click on didn't display the absolute date, so I stopped trying even though if I had tried it on the other places it would have worked... which I *eventually* discovered by accident.

Thanks for the tip!

**Comment by steveward** [comment]
Requested in https://github.com/desktop/desktop/issues/19154:

>Currently GitHub Desktop will helpfully show "2 months ago" for example, and if you hover your mouse over you'll see the full date. However, there's cases where I always want to see the full date in the user interface. Hovering my mouse over the dates just slows things down.

**Comment by jclark-dot-org** [comment]
A big use case for git in general and github desktop in particular is being able to find a specific change to the code base; Github Desktop is great for scanning a list of commits and quickly looking at the details.  

A use case that happens frequently for me is trying to quickly find a commit based on a specific date or time range.  I often have clients ask about a specific change from months ago; I can quickly find the date by searching email or issue-tracker tickets.  Many clients know the dates where changes were tested or went live.  But in an active repo, knowing you are looking for a commit from June 13 means scanning everything that says "2 months ago" or "3 months ago", because today is Aug 28 and I don't know if that's considered 2 or 3 months.  There might be several dozen commits over those two months.

But my human eyes can spot 2024-06-13 (or 6/13/2024, or Jun 13, 2024) out of sorted list of dates almost instantly.

**Comment by jessetan** [comment]
For the GitHub website there are three Chrome extensions that change the datetime display to be absolute:
- https://chromewebstore.google.com/detail/github-exact-dates/kjhknbbgbadlogbdfffgmngdgclomfol - https://github.com/kubilayeldemir/github-exact-dates
- https://chromewebstore.google.com/detail/github-exact-time/fanolkboeldmhkpeaanhhkogoeenbgko
- https://chromewebstore.google.com/detail/github-absolute-dates/iepecohjelcmdnahbddleblfphbaheno - https://github.com/egyptianbman/chrome-github-absolute-dates

**Comment by Jamie-Landeg-Jones** [comment]
Thanks for that. Unfortunately, I could only get one to work, and it only works on the commit list not the tree browsing.

The fact that people wrote this extension in the first place speaks volumes, yet no-one is listening.

**Comment by ledgersteve** [comment]
This ticket is two years old, lots of people are asking for it, and it is still not planned.  The issue that people are really clamouring for (see also all the tickets that have been merged into this and therefore closed) is to change '29 days ago', 'last month', '7 months ago' etc into something more meaningful, i.e. the actual commit date and time.  This of its own would seem to be not too hard a thing to achieve.  

It's sub-optimal then that it got rolled-up into a multi-issue ticket with things that are harder to achieve and less in demand.  I'd like to suggest that we split this ticket out into multiple single-issue tickets, and then we can think about prioritising the commit-date issue without having to implement the other issues also, which are less in demand, harder to implement, and not universally favoured.   With one issue per ticket, each issue can be considered on its own merits.  Hopefully the meaningful commit date request would then come to the top of the to-do list.

**Comment by nielsduelund** [comment]
+1 from me.
You have the actual date in the HTML, so it can be displayed when hovering.
Why not add a setting, so the user (un)friendly can be replaced with proper date and time.
It will not require *just* that, because changing format for the date and time should be added as well - IMHO.
It must however be a minor change compared to the wast possibilities availablen and the complicated merging algoritms etc.
/niels

**Comment by drstevens** [comment]
Apologies for the additional comment here, but I've put up with these "friendly" timestamps since 2010 and have finally worked up enough frustration to look for an official solution.

I _never_ want the relatively "friendly" time/date displayed. It's hard for me to imagine anyone would prefer "yesterday" instead of an actual timestamp, but I do understand that some people do. It seems like a user setting is necessary.

The hover behavior mostly works, but sometimes you just want to copy/paste the timestamp into something else, like a Grafana dashboard or JIRA issue. This is difficult/impossible with the hover workaround.

I know there are workarounds, like tampermonkey or bookmarkets, but really we shouldn't need to look for third-party workarounds that could potentially increase supply-chain attack surface area.

**Comment by aterbo** [comment]
@drstevens  I don't think you should feel the need to apologies for the comment, especially a relevant and clearly thought out one. Multiple times this and other similar issues were closed due to apparent lack of interest or demand from users. I think the fact that comments on these issues just keep stacking up is the only way that demand is shown...

**Comment by pablott** [comment]
I also think this is an issue. And is not just in the history, but also in the commit view. 
This does not really tell me much...
<img width="246" alt="Screenshot 2024-10-26 at 9 33 32" src="https://github.com/user-attachments/assets/3556435e-5415-4cb3-881b-f62b3d587e48">
Currently even hovering in such situation shows the actual date, so that seems to be broken or removed.

**Comment by jeffchensonos** [comment]
Any developer or PM here? I hate to push you, but it's my nightmare the history doesn't show exact date!

**Comment by jeffchensonos** [comment]
@sergiou87 Could you take a look at this? I don't know who is in charge of this request or who will care about this request, but I hope someone can address this. Thank you.

**Comment by HughWarrington** [comment]
> It's sub-optimal then that it got rolled-up into a multi-issue ticket with things that are harder to achieve and less in demand. I'd like to suggest that we split this ticket out into multiple single-issue tickets, and then we can think about prioritising the commit-date issue without having to implement the other issues also, which are less in demand, harder to implement, and not universally favoured. With one issue per ticket, each issue can be considered on its own merits. Hopefully the meaningful commit date request would then come to the top of the to-do list.

I absolutely agree.

Any of #5395, #9147, #17456, #17690, #17702, #17828, #17976, #18136, #18273, #18442, #18443, #18959, #19154, #19469, #19549 would do.

I'm not sure why those short and specific tickets have been closed in favour of this one, which is ill-defined and mixes several changes.

**Comment by cfrupp** [comment]
Heck yeah. Lumping all the issues together is counterproductive. There's this new idea, anybody heard of it? Agile development?

**Comment by antonydandrea** [comment]
To me this is such a quick thing to fix. In what universe is "Last week" more useful than the actual date.

**Comment by ICEknigh7** [comment]
Maybe reopen this: https://github.com/desktop/desktop/issues/9147

**Comment by RobinSchmidt** [comment]
Yes. I fully agree with this. An actual timestamp is *much* more useful than this vague indication. What really irks me about this is that there must be some code sitting somewhere in the system that has been written to purposefully and deliberately blur this (perfectly fine) raw timestamp information for supposedly "friendlier" GUI display (which it is really not) instead of passing it just straight onto the GUI. Just leave the info as is and display it straight. No need to dumb it down for display. This is not the right place for blurring information in a colloquial way. Git users are (typically) programmers, after all - so we can deal with the real thing. Or at least, make the timestamp blurring optional. Sorry for the author of the timestamp mangling algorithm - I don't mean to rude - even though I realize to possibly come across as such. I really like GitHub and GitHub Desktop. But that feature is really a disimprovement over just showing us the raw data as is.

**Comment by qneill-sifive** [comment]
To summarize in bold letters

# An actual timestamp is _much_ more useful than this vague indication

### An industry standard UI is _not_ the right place for "blurring information"
### just show the raw timestamp
### An option to switch between the two would be great
### If you wish, you can poll your users and come back with statistics to tell us we're all wrong in this thread
### But meanwhile add the option please


# This particular issue seems to be a recurrent request
(h/t @HughWarrington)

### see https://github.com/desktop/desktop/issues/5395, https://github.com/desktop/desktop/issues/9147, https://github.com/desktop/desktop/issues/17456, https://github.com/desktop/desktop/issues/17690, https://github.com/desktop/desktop/issues/17702, https://github.com/desktop/desktop/issues/17828, https://github.com/desktop/desktop/issues/17976, https://github.com/desktop/desktop/issues/18136, https://github.com/desktop/desktop/issues/18273, https://github.com/desktop/desktop/issues/18442, https://github.com/desktop/desktop/issues/18443, https://github.com/desktop/desktop/issues/18959, https://github.com/desktop/desktop/issues/19154, https://github.com/desktop/desktop/issues/19469, and https://github.com/desktop/desktop/issues/19549


# Moreover, it seems to have problems

### see https://github.com/desktop/desktop/issues/18009 where "1d ago" seems to be just wrong

**Comment by marwaygreg** [comment]
I want actual dates as well. Hovering to get a real date is a waste of time when scanning a list and wanting to find a date or understand the real dates.

Also dates from February 2 through March 5 reported as "3 months ago" is a useless generalization to me. Why this can't be a simple user config switch given the feedback is puzzling.

**Comment by cfrupp** [comment]
The GitHub Desktop GUI is unusable if you are doing anything with file history. I've given up on Microsloth doing anything about it. I use the command line. 

Git Bash is pretty good, and you can configure GHD to open it from the Repository > Open menu option.

**Comment by cmcculloh-kr** [comment]
If for some reason it is required for the UI to do the cognitive work of calculating "how long ago" a timestamp is, hide _that_ away in the title instead of the timestamp. Just _reverse_ what the current convention is:

![Image](https://github.com/user-attachments/assets/1faf502f-6e3a-4d4a-9316-1364f0308af9)

If the real concern here is something like "space" or "timestamps are ugly" at the very least you could show something like:

![Image](https://github.com/user-attachments/assets/8f142d73-0d78-4a16-805c-b592f0012904)

**Comment by rickydazla** [comment]
A toggle/config option here would be most welcome! Every few months I check and sigh. ✔️Just Do It

**Comment by pablott** [comment]
Since oddly GH seems to be ignoring this very reasonable request, meanwhile I'm using this extension (only for web version): https://chromewebstore.google.com/detail/enhance-github-timestamps/jpfkamffgggihjpgijeolhhmhjglcabd
It does what it says without frills.

**Comment by wmdash92** [comment]
It would be helpful to have an option to show commit dates in YYYY-MM-DD format instead of relative time. Absolute dates make tracking history easier.

**Comment by tphinney** [comment]
Mixing up a bunch of feature requests in one is irritating.

Just give me an option to ALWAYS see the actual date and time of every commit, everywhere it is displayed. This one missing feature makes me hate Github (which is otherwise quite lovely).

**Comment by cfrupp** [comment]
Do you think any Microsoft employee ever looks at this thread? Hello? Hellohhhh???

**Comment by zsolt-sb** [comment]
+1

**Comment by jottevanger** [comment]
please do this. Actual dates, and if you feel like getting fancy make it optional but otherwise just actual dates/times

**Comment by meyerjoe** [comment]
+1 
the "friendly" date format is totally useless

**Comment by SuperSoldado** [comment]
Why this nice option like this is "not planned"? The ticket is from 2022. In Github Desktop changelog I see lot's of useless changes nobody asked for. Have this project a leader? Look's like not.  Man I really miss GitKraken sadly this decent IDE is payed. I'll consider paying since this GitHub Desktop has bad support.

**Comment by SuperSoldado** [comment]
> +1 the "friendly" date format is totally useless

Like the support of this tool.

**Comment by ledgersteve** [comment]
> Why this nice option like this is "not planned"? The ticket is from 2022. 

@SuperSoldado Further info: Yes this ticket is from 2022, but the issue it describes is the subject of a lot of other request tickets too, the earliest of which is from August 2018. #5395, #9147, #14611, #17456, #17690, #17702, #17828, #17976, #18136, #18273, #18442, #18443, #18959, #19154, #19469, #19549

**Comment by SuperSoldado** [comment]
@ledgersteve Seems like what the users wants differs to what developers wants to do. See this last improvement: 

<img width="703" height="105" alt="Image" src="https://github.com/user-attachments/assets/c51e879f-bd2a-43f5-bc6d-072865d01661" />

Who really cares to this? I'll try to contact someone in high level, above the project manager. Crying here has no use.

**Comment by agarny** [comment]
Here as a result of that "friendly" date format which has been annoying me for years (same with the GitHub website) and... I have just noticed that if we move our mouse pointer over any of those "friendly" dates then we can actually see the date.

Still, this means that we have to move the mouse, wait, check the "hint" and go to the next "friendly" date. So, it's not at all convenient when you need to check commits from "2 months ago" which corresponds to several weeks of commits...

**Comment by ICEknigh7** [comment]
> we have to move the mouse, wait, check the "hint" and go to the next "friendly" date. So, it's not at all convenient when you need to check commits from "2 months ago" which corresponds to several weeks of commits...

And you can't hover on mobile.

**Comment by antonydandrea** [comment]
> > we have to move the mouse, wait, check the "hint" and go to the next "friendly" date. So, it's not at all convenient when you need to check commits from "2 months ago" which corresponds to several weeks of commits...
> 
> And you can't hover on mobile.

Exactly! This is the crux of the problem for me, I can get over the Hover on desktop, but on my phone I can not see the date at all.

Overall the question still hasn't been answered, who finds "2 months ago" more useful or readable than a date. GitHub is primarily used by developers!

**Comment by MartinDawsonHalff** [comment]
Yes, this, _please._ It's bad enough getting this vague "x ago" in social media, where it might be kind of relevant that you're viewing an old post, but when I'm looking at a list of commits, I can't imagine any case where I'd want to know how many months ago it was rather than the date it was committed on.

**Comment by cfrupp** [comment]
I would really love to know if any Microsoft employee ever reads this thread. Or have you all unsubscribed? Anyone?

**Comment by tinu1976ch** [comment]
> I would really love to know if any Microsoft employee ever reads this thread. Or have you all unsubscribed? Anyone?

@cfrupp  why should *Microsoft* read that?

**Comment by SuperSoldado** [comment]
> I would really love to know if any Microsoft employee ever reads this thread. Or have you all unsubscribed? Anyone?

I tweeted Thomas Donke (@ashtom supposed to be the boss) some time ago. He did nothing as expected. Seems GitHub is a ship having no clear leadership. I used to read the changelogs but stopped 1 year ago because not much relevant stuff FOR ME was released. Seems last features are more related to what marketing wants then what user's wants. They are probably including some useless A.I. feature nobody asked for just make the app heavier, spend more resources and bother people with useless hints.

**Comment by cfrupp** [comment]
> > I would really love to know if any Microsoft employee ever reads this thread. Or have you all unsubscribed? Anyone?
> 
> [@cfrupp](https://github.com/cfrupp) why should _Microsoft_ read that?

They own GitHub, and are responsible for its support.

Or irresponsible for its support, as the case may be.

**Comment by shou-gana1** [comment]
> > > I would really love to know if any Microsoft employee ever reads this thread. Or have you all unsubscribed? Anyone?
> > 
> > 
> > [@cfrupp](https://github.com/cfrupp) why should _Microsoft_ read that?
> 
> They own GitHub, and are responsible for its support.
> 
> Or irresponsible for its support, as the case may be.

unfortunately, most companies are kinda infamous for not really doing anything about any support or feedback they receive even despite that fact, so 🤷🏾

**Comment by procura-interna** [comment]
Eclipse/MeGit also has the "relative date" behaviour by default, but you can change it in the settings to show absolute dates. It's one of the first things I change after a fresh install.

I would love to have the option to opt-out of relatives dates on GitHub desktop too, as its the tool I use most for git nowadays, as I am often trying to match commits with calendar days and having to mouse over commits all the time to convert the relative date to an absolute date is a bit frustrating.

I gave this issue a "thumbs up" only because of the date display change. I never find myself wanting to see commit IDs on the history list.

**Comment by giduac** [comment]
Current time display option is next to useless in bigger repos.

The time stamp will be a great addition and definitely add to the usability.
Coupled with a filter on date time would even make it better.

**Comment by dlwiii** [comment]
Relative dates are never helpful to me. I would love to see the exact dates in the list of commits.  Not in the mouseovers. And UTC would be fine.  It seems odd that the relative date is sued since it actually seems harder from a UI point of  view.  We have alot of commits in our repo and I like to scan things quickly.

**Comment by dj707chen** [comment]
It's a shame that relative dates are displayed by default and no way to change that, relative dates are for social media postings.

**Comment by cfrupp** [comment]
I suspect it is useless to post on this thread. This issue is unassigned and tagged "not-planned".

<img width="189" height="131" alt="Image" src="https://github.com/user-attachments/assets/06ccf636-0bf4-47dd-bd26-95fcf2a88de2" />

I just started a new issue https://github.com/desktop/desktop/issues/21318.

**Comment by RobinSchmidt** [comment]
> I suspect it is useless to post on this thread. This issue is unassigned and tagged "not-planned".

Well - after all, the app is open source. Some outsider could perhaps just do it themselves and issue a pull request. This is the great thing (or one of the great things) about open source, right?

> relative dates are for social media postings.

I'd rather see unblurred dates there, too. Maybe in the web browser history it may make some more sense. But there, really, only for the purpose of binning site visits into "yesterday", "last week", etc. Within each bin, I'd also prefer to see actual dates.

**Comment by beermattuk** [comment]
I've no idea why it's like this in the first place.  This isn't Facebook or some tool for halfwits, it's a professional tool for people who want attention to detail.

If you can't work out that today's date -1 is yesterday, then you're in the wrong job.

Always show the exact datetime.

**Comment by joshjeff-stripe** [comment]
@steveward Any update on this, three years on? The "friendly" date time continues to be rather the opposite of useful

**Comment by katzlbt** [comment]
If you can work out that "last month" and 2026-01-12 tag in your code are the same you are a freaking superhuman!
> If you can't work out that today's date -1 is yesterday, then you're in the wrong job.

Please fix this facebook style crap! I am praying for this since years.

**Comment by ICEknigh7** [comment]
OP is now a ghost. 😱

**Comment by HansHighness** [comment]
It should be easy to fix, which makes it even more annoying. And while you’re at it, throw the AM/PM format back to the 1800s.

**Comment by RobinSchmidt** [comment]
Yay! It has been done! Very good! Thanks!

**Comment by rickydazla** [comment]
I don't think this has made it to production yet 🤔

**Comment by niik** [comment]
Hey @rickydazla, you're correct. We've shipped this to our beta channel for some testing before we promote it to production. We'd love it if you could download our [beta channel](https://desktop.github.com/beta/) and give it a go and provide any feedback you might have! That'll help raise our confidence in shipping this to a wider audience.

**Comment by ICEknigh7** [comment]
@niik Please also fix it on the website.

**Comment by zsolt-sb** [comment]
> [@niik](https://github.com/niik) Please also fix it on the website.

Calling out there have been a few bugs that don't specifically ask for this on the desktop app, but make a generic appeal.  I have been on a few of them before they were closed out to this bug and I believe some of the comments on these assumed this was a fix we were also asking for on the web.

**Comment by antonydandrea** [comment]
How is this not going to be fixed on the website?! YEARS after asking for it, its finally agreed to be fixed SOMEWHERE, surely logically you agree its a problem then, and will fix everywhere else.

Again, I can't fathom how anyone can defend unprecise times over precise date and time.

**Comment by niik** [comment]
For suggestions and feedback regarding GitHub products other than the open source GitHub Desktop project please engage in the [GitHub community forums](https://github.com/orgs/community/discussions) instead.

For feedback regarding this specific feature of GitHub Desktop we'd welcome comments on the PR, or in a new issue (given the number of comments and people already watching this issue).

**Comment by rickydazla** [comment]
🤩 @niik praise the gods! It's so beautiful. Thanks for the heads-up and beta link. If we have feedback, which PR should it go on, this closed one https://github.com/desktop/desktop/pull/21836 ?

**Comment by niik** [comment]
@rickydazla Yup, that PR is great, or if you think your feedback warrants it you may open a new issue 👍

**Comment by ICEknigh7** [comment]
Oh my god, they really are only fixing the app and not the website, aren't they.

**Comment by antonydandrea** [comment]
> For suggestions and feedback regarding GitHub products other than the open source GitHub Desktop project please engage in the [GitHub community forums](https://github.com/orgs/community/discussions) instead.
> 
> For feedback regarding this specific feature of GitHub Desktop we'd welcome comments on the PR, or in a new issue (given the number of comments and people already watching this issue).

So you're ignoring hundreds of comments on this thread. Took years to take some action and its only half the job. OK. Great job.

Still no explanation or justification for the unuseful format of the dates and yet still not fix.

**Comment by GumbyXGames** [comment]
Can't we just reopen the ticket? I get they will probably close it but if it's reopened as not fixed maybe they will pay attention for once.

**Comment by tphinney** [comment]
This is fixed! At least in the web interface. You probabky need to adjust your preferences though — pretty sure it is an option, not on by default.

**Comment by jottevanger** [comment]
it is indeed fixed in Desktop, hooray! Under Options > Appearance > Formatting check the box for "Prefer absolute dates over relative". This should of course be ticked by default but I'm nit-picking now

**Comment by GumbyXGames** [comment]
> it is indeed fixed in Desktop, hooray! Under Options > Appearance > Formatting check the box for "Prefer absolute dates over relative". This should of course be ticked by default but I'm nit-picking now

I don't have that option

<img width="1080" height="2203" alt="Image" src="https://github.com/user-attachments/assets/809ab648-7ca7-4d3b-9aba-885bd6795a26" />

**Comment by Taranchul** [comment]
> This is fixed! At least in the web interface. You probabky need to adjust your preferences though — pretty sure it is an option, not on by default.

It doesn't seem to be, at least not under Settings > Appearance. Did I miss it somewhere else?

> This should of course be ticked by default but I'm nit-picking now

Definitely. These relative dates strike me as very "outlookesque" and like a "feature" that's trickled down from Github's owner, Microsoft. Such translation to everyday language might be well-meant for normal users (and IMHO even arguable there), but I suspect that they're mostly useless and maybe even irritating for Github's more technical-minded user base. They certainly are to me.
